spec/unit/helpers/check_helper_spec.rb

```diff
@@ -38,18 +43,78 @@
   end
 
   it 'checks types' do
-    PARAMS_3 = GoodData::LCM2::BaseAction.define_params(self) do
-      description 'Testing param four'
-      param :test_param_four, instance_of(GoodData::LCM2::Type::StringType), required: false
-    end
-    expect { GoodData::LCM2::Helpers.check_params(PARAMS_3, params) }.to raise_error(/has invalid type/)
+    params['test_param_one'] = 'test_param_one'
+    expect { GoodData::LCM2::Helpers.check_params(spec, params) }.to raise_error(/has invalid type/)
   end
 
   it 'fails when unspecified variable is acessed' do
-    params.setup_filters(PARAMS_3)
+    params.setup_filters(spec)
     expect { params[:test_param_three] }.to raise_error(/not defined in the specification/)
   end
 
+  context 'when created from stringified hash' do
+    let(:params) do
+      raw_params = { 'update_preference' => { 'keep_data' => false,
+                                              'allow_cascade_drops' => true } }
+      GoodData::LCM2.convert_to_smart_hash(raw_params)
+    end
+
+    let(:other) do
```

Comment:
It would be prettier to move this `let` to a context with the examples that use it. It would also allow us to use RSpec's `#subject` instead of typing `GoodData::LCM2::Helpers.check_params(spec, params)` every time.